lib/functions.ts

```diff
@@ -11,3 +13,7 @@ export function assert(condition: any, msg: string = "no additional info provide
 }
 
 export function noop(..._args: unknown[]): void {}
+
+export function isExact<T, EXACT_SHAPE>(x: Exact<T, EXACT_SHAPE>): EXACT_SHAPE {
+  return x;
+}
```

Comment:
I found one workaround which help get rid of `typeof obj`: we create curried version of the function, but use only the first generic type (second will be inferred automatically) – https://github.com/krzkaczor/ts-essentials/pull/248

If I find something else, I will let you know

Feel free to merge or to close it

Reply:
Merged too ✅